### ISSUE TITLE
Add duoBERT

### ISSUE THREAD

**Issue by ronakice** [castorini/pygaggle#15]
I was looking into adding duo support (k-way ranking potentially) for MSMARCO and TREC-CAR. 

For RelevanceExample, I was wondering if we can have documents as Union(List[Text], List[List[Text]]) instead of just List[Text]. I could also do something like this from the evaluate method in RerankerEvaluator. Another way is to add an is_duo argument to the various reranks/class. We could also create different duo rankers all together. 

Thoughts?

**Comment by rodrigonogueira4** [comment]
Yeah, that sounds good to me! For now, I would worry only about duo and forget about the general case as we don't even know how to make it computationally viable

**Comment by ronakice** [comment]
@wiltan-uw is on this!

**Comment by lintool** [comment]
I'm wondering what the relationships between this issue and this repo is: https://github.com/castorini/duobert

Are we planning on deprecating the `duobert` repo, i.e., but into archive real-only mode?

**Comment by rodrigonogueira4** [comment]
Yeah, I think archiving duobert repo is a good idea. It is in tensorflow, and it requires some manual steps to do reranking
